pkg/operator/ceph/webhook-config.go

```diff
@@ -0,0 +1,297 @@
+	_, err := clusterdContext.Clientset.AdmissionregistrationV1().ValidatingWebhookConfigurations().Get(ctx, webhookConfigName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return createValidatingWebhookConfig(ctx, clusterdContext)
+		}
+		logger.Errorf("failed to get webhook %s. %v", webhookConfigName, err)
+	}
+
+	return nil
+}
+
+func createValidatingWebhookConfig(ctx context.Context, clusterdContext *clusterd.Context) error {
+	sideEffects := admv1.SideEffectClassNone
+	var timeout int32 = 5
+	serviceCephClusterPath := "/validate-ceph-rook-io-v1-cephcluster"
+	serviceCephBlockPoolPath := "/validate-ceph-rook-io-v1-cephblockpool"
+	serviceCephObjectStorePath := "/validate-ceph-rook-io-v1-cephobjectstore"
+
+	logger.Infof("Creating webhook %s/%s.", namespace, webhookConfigName)
+
+	validatingWebhook := &admv1.ValidatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      webhookConfigName,
+			Namespace: namespace,
+			Annotations: map[string]string{
+				"cert-manager.io/inject-ca-from": fmt.Sprintf("%s/rook-admission-controller-cert", namespace),
+			},
+		},
+		Webhooks: []admv1.ValidatingWebhook{
+			{
+				Name: fmt.Sprintf("cephcluster-wh-%s-%s.rook.io", admissionControllerAppName, namespace),
+				Rules: []admv1.RuleWithOperations{
+					{
+						Rule: admv1.Rule{
+							APIGroups:   []string{"ceph.rook.io"},
+							APIVersions: []string{"v1"},
+							Resources:   []string{"cephclusters"},
+						},
+						Operations: []admv1.OperationType{
+							admv1.Update,
+							admv1.Create,
+							admv1.Delete,
+						},
+					},
+				},
+				ClientConfig: admv1.WebhookClientConfig{
+					Service: &admv1.ServiceReference{
+						Name:      admissionControllerAppName,
+						Namespace: namespace,
+						Path:      &serviceCephClusterPath,
+					},
+				},
+				AdmissionReviewVersions: []string{"v1", "v1beta1"},
+				SideEffects:             &sideEffects,
+				TimeoutSeconds:          &timeout,
+			},
+			{
+				Name: fmt.Sprintf("cephblockpool-wh-%s-%s.rook.io", admissionControllerAppName, namespace),
+				Rules: []admv1.RuleWithOperations{
+					{
+						Rule: admv1.Rule{
+							APIGroups:   []string{"ceph.rook.io"},
+							APIVersions: []string{"v1"},
+							Resources:   []string{"cephblockpools"},
+						},
+						Operations: []admv1.OperationType{
+							admv1.Update,
+							admv1.Create,
+							admv1.Delete,
+						},
+					},
+				},
+				ClientConfig: admv1.WebhookClientConfig{
+					Service: &admv1.ServiceReference{
+						Name:      admissionControllerAppName,
+						Namespace: namespace,
+						Path:      &serviceCephBlockPoolPath,
+					},
+				},
+				AdmissionReviewVersions: []string{"v1", "v1beta1"},
+				SideEffects:             &sideEffects,
+				TimeoutSeconds:          &timeout,
+			},
+			{
+				Name: fmt.Sprintf("cephobjectstore-wh-%s-%s.rook.io", admissionControllerAppName, namespace),
+				Rules: []admv1.RuleWithOperations{
```

Comment:
In a follow up PR it would be nice to have a helper method for adding each of these admission controllers to the slice since most of the properties are the same.

Reply:
https://github.com/rook/rook/issues/10025